config/deploy.rb

```diff
@@ -88,42 +100,50 @@
         --env OSS_ROOT_URL=#{fetch(:oss_root_url)} \
         -P --name slidehub#{prefix} ryuzee/slidehub
 EOS
-      container_id = capture("sudo #{cmd}")
-      port = capture("sudo docker port #{container_id} 3000").to_s.split(':')[1]
-      puts port
-      # confirm running
-      cmd = "curl -LI http://127.0.0.1:#{port} -o /dev/null -w '%{http_code}\\n' -s"
-      cnt = 0
-      loop do
-        execute "echo 'sleep 20 sec...'"
-        sleep 20
-        cnt += 1
-        if cnt == 10
-          error = Exception.new('An error that should abort and rollback deployment')
-          raise error
-        end
-        begin
-          container_status = capture(cmd).to_i
-          if container_status == 200
-            break
-          end
-        rescue Exception => e
-          puts e.inspect
-        end
-      end
-      data = { port: port }
-      template 'nginx_default.erb', '/tmp/default', data, true
-      execute 'sudo mv /tmp/default /etc/nginx/sites-available/default && sudo service nginx reload'
-
-      containers = capture('sudo docker ps -q').to_s.split("\n")
-      containers.shift
-      containers.shift
-      puts containers.inspect
-      containers.each do |c|
-        execute "sudo docker rm -f #{c}"
+      run_app(cmd)
+      clean_container
+    end
+  end
+end
+
+def run_app (docker_cmd)
```

Comment:
__[rubocop]__
* Metrics/MethodLength: Method has too many lines. [26/20] [:ref](https://github.com/bbatsov/ruby-style-guide#short-methods)
* Style/SpaceAfterMethodName: Do not put a space between a method name and the opening parenthesis. [:ref](https://github.com/bbatsov/ruby-style-guide#parens-no-spaces)